packages/editor/src/index.ts

```diff
@@ -7,53 +7,66 @@ import {
   usePrefix,
   isEnterKey,
   AsyncPromptConfig,
+  InquirerReadline,
 } from '@inquirer/core';
 
 type EditorConfig = AsyncPromptConfig & {
   default?: string;
   postfix?: string;
+  waitUserInput?: boolean;
 };
 
-export default createPrompt<string, EditorConfig>((config, done) => {
+let firstRender = true;
+export default createPrompt<string, EditorConfig>((config, done, rl) => {
   const [status, setStatus] = useState<string>('pending');
   const [value, setValue] = useState<string>(config.default || '');
   const [errorMsg, setError] = useState<string | undefined>(undefined);
 
+  const startExternalEditor = (rl: InquirerReadline) => {
+    rl.pause();
+    editAsync(
+      value,
+      async (error, answer) => {
+        rl.resume();
+        if (error) {
+          setError(error.toString());
+        } else {
+          setStatus('loading');
+          const isValid = await config.validate(answer);
+          if (isValid === true) {
+            setError(undefined);
+            setStatus('done');
+            done(answer);
+          } else {
+            setValue(answer);
+            setError(isValid || 'You must provide a valid value');
+            setStatus('pending');
+          }
+        }
+      },
+      {
+        postfix: config.postfix || '.txt',
+      }
+    );
+  };
+
   useKeypress(async (key, rl) => {
     // Ignore keypress while our prompt is doing other processing.
     if (status !== 'pending') {
       return;
     }
 
     if (isEnterKey(key)) {
-      rl.pause();
-      editAsync(
-        value,
-        async (error, answer) => {
-          rl.resume();
-          if (error) {
-            setError(error.toString());
-          } else {
-            setStatus('loading');
-            const isValid = await config.validate(answer);
-            if (isValid === true) {
-              setError(undefined);
-              setStatus('done');
-              done(answer);
-            } else {
-              setValue(answer);
-              setError(isValid || 'You must provide a valid value');
-              setStatus('pending');
-            }
-          }
-        },
-        {
-          postfix: config.postfix || '.txt',
-        }
-      );
+      startExternalEditor(rl);
     }
   });
 
+  if (!(config.waitUserInput ?? true) && firstRender && rl !== undefined) {
+    startExternalEditor(rl);
+    firstRender = false;
+    return '';
+  }
```

Comment:
Let's put this into a `useEffect()` handler instead. It doesn't change anything now, but I think that'll be more in line with potential future optimizations.